### ISSUE TITLE
Unused variable not detected when also set inside loop

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1626]
```php
function foo() : void {
  $a = 1;

  if (rand(0, 1)) {
    while (rand(0, 1)) {
      $a = 2;
    }
  }
}
```

Expected: UnusedVariable
Actual: No error

https://psalm.dev/r/506158b447

cc @TysonAndre, fixed in Psalm's code in #1624
